src/routers/controllers/groups.controllers.ts

```diff
@@ -0,0 +1,56 @@
+
+import {
+    ValidatedRequest
+} from 'express-joi-validation';
+import Group from '../../models/groups/group.types';
+import { GroupRequestSchema } from '../../models/groups/groups.interface';
+import { v4 as uuidv4 } from 'uuid';
+import GroupsServices from '../../services/groups.service';
+
+export class GroupsController {
+    async getAll(req:any, res:any) {
+        return res.json(await GroupsServices.getAllGroups());
+    }
+
+    createGroup(req: ValidatedRequest<GroupRequestSchema>, res:any) {
+        const group: Group = {
+            group_uid: uuidv4(),
+            ...req.body
+        };
+        GroupsServices.createGroup(group);
```

Comment:
promise from function is ignored, so you return result before process is completed

Reply:
done